src/shared/components/msktable/MSKTable.tsx

```diff
@@ -9,16 +9,37 @@ import {CopyDownloadControls} from "../copyDownloadControls/CopyDownloadControls
 import {serializeData} from "shared/lib/Serializer";
 import DefaultTooltip from "../DefaultTooltip";
 import {ButtonToolbar} from "react-bootstrap";
+import {numberSort, sortBy, numberListSort, stringListSort, stringSort} from "../../lib/SortUtils";
+import * as _ from "lodash";
 
 export type Column<T> = {
     name: string;
     filter?:(data:T, filterString:string, filterStringUpper?:string, filterStringLower?:string)=>boolean;
-    sort?:(a:T, b:T, ascending:boolean)=>number;
     visible?:boolean;
     render:(data:T)=>JSX.Element;
     download?:(data:T)=>string;
     tooltip?:JSX.Element;
-};
+} & ({
+    sort?:undefined,
+    sortValue:(data:T)=>number|null;
+    sortType:"number";
+} | {
+    sort?:undefined,
+    sortValue:(data:T)=>(number|null)[];
+    sortType:"number[]";
+} | {
+    sort?:undefined,
+    sortValue:(data:T)=>string|null;
+    sortType:"string";
+} | {
+    sort?:undefined,
+    sortValue:(data:T)=>(string|null)[];
+    sortType:"string[]";
+} | {
+    sort?:(a:T, b:T, ascending:boolean)=>number
```

Comment:
if you have complicated rules about what comes before what which aren't easily mapped to a numerical/string value? maybe unnecessary